### DISCUSSION TITLE
Using the method of defining parameters first and then passing them into the function makes it impossible to use extended fields.

### DISCUSSION THREAD

**Discussion by lycpan233** [prisma/prisma#25531] [Q&A]
### Bug description

I might be doing some business before the query, so I want to define an object before passing it into the function, but the input type I exported doesn't support extended fields.

### How to reproduce

1. Expand the client result field
2. Use definitions to export types and pass in function
3. Get ` error TS2353: Object literal may only specify known properties, and 'xx' does not exist in type 'xxxSelect<DefaultArgs>'.`

### Expected behavior

I hope to get some examples of this.

### Prisma information

```prisma
model User {
    id           Int     @id @default(autoincrement())
    nickname     String? @db.VarChar(30)
    email        String? @db.VarChar(50)
    password     String  @default("") @db.VarChar(64)
    salt         String  @default("") @db.VarChar(32)
    roles        String? @default("|ROLE_USER|") @db.VarChar(255)

    @@map("user")
}

```

```ts
import { Prisma, PrismaClient } from '@prisma/client';

const prisma = new PrismaClient().$extends({
  result: {
    user: {
      rolesArr: {
        needs: { roles: true },
        compute(user) {
          return user.roles.split('|').filter(item => item !== '');
        },
      },
    },
  },
});

async function main() {
  const options: Prisma.UserFindFirstArgs = {
    select: {
      rolesArr: true,// error TS2353: Object literal may only specify known properties, and 'rolesArr' does not exist in type 'UserSelect<DefaultArgs>'.
    },
  };
  const user = await prisma.user.findFirst(options);
  console.log(user);
  // but, This one can be used
  const user = await prisma.user.findFirst({
    select: {
      rolesArr: true,
    },
  });
}

main()
  .then(async () => {
    await prisma.$disconnect();
  })
  .catch(async e => {
    console.error(e);
    await prisma.$disconnect();
    process.exit(1);
  });

```

### Environment & setup

- OS: mac OS Sequoia 15.0.1
- Nodejs v20.14.0
- npm 10.7.0

### Prisma Version

```
prisma                  : 5.20.0
@prisma/client          : 5.20.0
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v20.14.0
Query Engine (Node-API) : libquery-engine 06fc58a368dc7be9fbbbe894adf8d445d208c284 (at node_modules/.pnpm/@prisma+engines@5.20.0/node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli 06fc58a368dc7be9fbbbe894adf8d445d208c284 (at node_modules/.pnpm/@prisma+engines@5.20.0/node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.20.0-12.06fc58a368dc7be9fbbbe894adf8d445d208c284
Default Engines Hash    : 06fc58a368dc7be9fbbbe894adf8d445d208c284
Studio                  : 0.502.0
Preview Features        : prismaSchemaFolder, strictUndefinedChecks
```

**Comment by jkomyno** [comment]
Hi @lycpan233, this is expected, as `Prisma.UserFindFirstArgs` is defined at "compile-time" (i.e., when you run `prisma generate` against your `schema.prisma` file). On the other hand, the types for the extended Prisma Client are inferred from your TypeScript file that imports the generated Prisma Client.

If you really want to have the `options` in a separate variable, you may want to play around with the `Parameters<typeof prisma.user.findFirst>` type utility from [TypeScript's standard library](https://www.typescriptlang.org/docs/handbook/utility-types.html#parameterstype).

**Reply by lycpan233** [reply]
Thank you for your reply. I tried the method you provided, but it still doesn't prompt to expand the field.
```ts
const options: Parameters<typeof prisma.user.findFirst>[0] = {
    select: {
      rolesArr: true,
    },
  };
  const user = await prisma.user.findFirst(options);
```

---
I met my expectations with the following example:

```ts
type ExtArgs = {
  result: {
    user: {
      rolesArr: true;
    };
  };

type UserFindFirstArgsExtends = Prisma.UserFindFirstArgs<InternalArgs & Extends>;
```
I'm a beginner at TS and don't know if there's a better way to practice it

**Reply by ludralph** [reply]
Hi @lycpan233 

You might want to check out this [Typescript lesson](https://www.totaltypescript.com/tutorials/beginners-typescript)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
